libraries/AP_Scripting/examples/mission-load.lua

```diff
@@ -0,0 +1,65 @@
+-- Example of loading a mission from the SD card using Scripting
+-- Would be trivial to select a mission based on scripting params or RC switch
+
+local function read_mission(file_name)
+
+  -- Open file
+  local file = io.open(file_name)
+  if not file then
+    error("Could open %s",file_name)
+  end
+
+  -- check header
+  local line = file:read('l')
+  if string.find(line,'QGC WPL 110') ~= 1 then
+    error(string.format("%s incorrect format",file_name))
+  end
+
+  -- clear any existing mission
+  if not mission:clear() then
+    error("Could not clear current mission")
+  end
+
+  -- read each line and write to mission
+  local item = mavlink_mission_item_int_t()
+  local index = 0
+  while true do
+
+    local data = {}
+    for i = 1, 12 do
+      data[i] = file:read('n')
+      if data[i] == nil then
+        if i == 1 then
+          gcs:send_text(6,string.format('loaded mission: %s',file_name))
+          return -- got to the end of the file
+        else
+          error('failed to read file')
+        end
+      end
+    end
+
+    item:seq(data[1])
+    item:frame(data[3])
+    item:command(data[4])
+    item:param1(data[5])
+    item:param2(data[6])
+    item:param3(data[7])
+    item:param4(data[8])
+    item:x(data[9]*10^7)
```

Comment:
In MAVLINK communication with GCS, we communicate with INT messages.
Floating values have a larger error than integer values.

MAVLINK WAYPOINT UPLOAD protocol:.
https://mavlink.io/en/services/mission.html#uploading_mission